### ISSUE TITLE
feature - avoid utf-8 decoding for text frames

### ISSUE THREAD

**Issue by toppk** [python-websockets/websockets#1376]
just because it is supposed to be in utf-8, doesn't mean I prefer it in that form.  specifically, my usecase, is giving the data to orjson, and passing it around as an orjson.Fragment().  

Here are the documents for that use case.

https://github.com/ijl/orjson#deserialize
https://github.com/ijl/orjson#fragment

looking at websockets code, if such a capability were to be implemented, it seems like we'd want to add an flag to WebSocketCommonProtocol() and then use it to force binary around the time it decides on whether to decode it or not, located here:

https://github.com/python-websockets/websockets/blob/main/src/websockets/legacy/protocol.py#L1053

I'd be happy to whip up a patch in case you would consider this feature request.

**Comment by toppk** [comment]
That would work quit well.  I guess I misunderstood the code, because it looked to me as if the recv() method is decoupled from where the actual processing of inbound data (read_message()).  The solution you propose would certainly be more flexible.

**Comment by toppk** [comment]
just thinking about the send side, I think it really is less important.  there aren't too many servers that are strict in what they accept, especially when they are expecting text.  I think if we implement it for send, while the effect is the same (skip encode, skip decode), but the names of the options will be different, e.g: decode_text_frames=False for recv(), and send_as_text=True for send()

**Comment by aaugustin** [comment]
Yes, we need to pick the names for both sides carefully and, ideally, consistently.

`raw_utf8` is a name that could work for both sides. I'm not sure it's the best name we can find, though.

If we have two names, I'd like some symmetry e.g. using the words `decode` and `encode`.

**Comment by carlos-sarmiento** [comment]
I'm finding myself in the same position, trying to send data encoded with `orjson` as a text frame even when it is provided to websockets in binary form.

Any chance this gets added?

**Comment by aaugustin** [comment]
This will be added as part of the new asyncio implementation (#1332).

**Comment by aaugustin** [comment]
The new asyncio implementations supports `recv(decode=False)`, which is the original request here.

(Also `recv(decode=True)` for the opposite behavior.)

I'm not planning to work on the other features discussed above, notably `send()`, until someone has a use case.

**Comment by aaugustin** [comment]
`send()` was handled in #1515.
